### PR TITLE
make verify command work in bundler

### LINKED ISSUE #36: Control whether `gem install` verifies signatures with an environment variable
In #34, we are letting users verify gem signatures as part of `gem install` as long as they set `--verify-signatures`. This is easier than explicitly `gem verify`ing installed gems, but it only works while using the `gem` client directly.  `bundle install` is left out.

Luckily, `bundle install` plays nice with rubygems plugins: the `pre_install` hook gets invoked as hoped.  `--verify-signatures` is out of the picture, but we can define an environment variable that `verify_extend` looks for in the absence of `--verify-signatures`.  Look at https://github.com/rubygems/rubygems/pull/4697 for inspiration.

A CI workload could install our plugin and also set the environment variable in order to verify the signatures of gems before installing them.

### PR DESCRIPTION
**Pull request by doodzik**
This PR fixes #36 and enables the usage of gem signature verifying with the usage of bundler.

I'm mainly copied the existing verify command and had to adjust the code a bit because I don't have the command class available. I also introduced a concept of `signing-policy`, which will enable us to provide different signing stories.

Running the gem command with the signing-policy env variable
Screen Shot 2021-12-16 at 11 01 51 AM

Running bundler with the --verbose flag and the signing-policy env variable
Screen Shot 2021-12-16 at 5 46 07 PM

I added a raise statement to the command to see if it would fail bundle install and it did.
Screen Shot 2021-12-16 at 5 48 51 PM

I tested the commands with the `gem` and `bundler` command and it works for both.

Feel free to merge this PR or make changes to it.

### CODE REVIEW

File: lib/rubygems/commands/verify_extend.rb [hunk 1]
```diff
@@ -25,13 +26,32 @@
...
-  begin
-    if (Gem::Sigstore.options[:verify])
-      puts "verify called"
+i.add_option('--rekor-host HOST', 'Rekor host') do |value, options|
+  options[:host] = value
+end
+
+class Gem::Commands::InstallCommand
+  alias_method :original_execute, :execute
+  def execute
```
Comment (rochlefebvre) [inline lib/rubygems/commands/verify_extend.rb:35]: Any particular reasons why we're not leveraging the `pre_install` hook for plugins?  I would only reopen the command class and redefine its entry method if necessary.

File: lib/rubygems/sigstore/security_policy.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class Gem::SecurityPolicy
```
Comment (rochlefebvre) [inline lib/rubygems/sigstore/security_policy.rb:1]: Security policy sounds very broad.  How about we start with Signing Policy?

File: lib/rubygems/commands/verify_extend.rb (hunk 1 above)
Comment (doodzik) [reply lib/rubygems/commands/verify_extend.rb:35]: I was checking if it makes a difference of the flag is implemented.

**Comment by doodzik** [comment]
As a followup to this PR and https://github.com/Shopify/ruby-sigstore/pull/51 we can consolidate the logic for the verify command into one spot.

File: lib/rubygems/commands/verify_extend.rb [hunk 3]
```diff
@@ -27,10 +27,25 @@
...
+        gemfile = Gem::Sigstore::Gemfile.new(gem_path)
+        verifier = Gem::Sigstore::GemVerifier.new(
+          gemfile: gemfile,
+          config: Gem::Sigstore::Config.read
+        )
+        verifier.run
+      end
     end
-  rescue Gem::SigstoreException => ex
+  rescue StandardError => ex
```
[the remark this answers is not in the document]
Comment (rochlefebvre) [reply lib/rubygems/commands/verify_extend.rb:48]: Yup, I have hit this same problem. I will create an issue to track the introduction of a proper exception hierarchy.

File: lib/rubygems/commands/verify_extend.rb [hunk 4]
```diff
@@ -27,10 +27,25 @@
...
   begin
-    if (Gem::Sigstore.options[:verify])
-      puts "verify called"
+    verify = Gem::Sigstore.options[:verify] || Gem::SigningPolicy.verify_gem_install?
+    if verify
+      # A locally installed gem will sometimes not have a reference to the .gem file
+      if (package = installer.package)
+        gem_path = package.gem.path
+
+        pp "Verifying #{gem_path}"
```
Comment (rochlefebvre) [inline lib/rubygems/commands/verify_extend.rb:36]: Let's use rubygems' own UI printing commands, like `say`.
```ruby
say "Verifying #{gem_path}"
```

File: lib/rubygems/commands/verify_extend.rb [hunk 5]
```diff
@@ -27,10 +27,25 @@
...
+
+        raise Gem::CommandLineError, "#{gem_path} is not a file" unless File.file?(gem_path)
```
Comment (rochlefebvre) [inline lib/rubygems/commands/verify_extend.rb:38]: Decent placeholder until we address #37.

**Review by rochlefebvre — APPROVED** [review]

**Review by jchestershopify — APPROVED** [review]
LGTM.

**Pull request merged**